bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1201,6 +1227,40 @@ public ByteBuf getExplicitLac(long ledgerId) throws IOException, Bookie.NoLedger
         return lac;
     }
 
+    /**
+     * Force sync given 'ledgerId' entries on the journal to the disk.
+     * This is useful for ledgers with DEFERRED_SYNC write flag
+     * for which the client receives acknoledgement for writes
+     * without the guarantee that data has been sync'd to disk (ackBeforeSync=true).
+     */
+    public void forceLedger(long ledgerId, WriteCallback cb,
+                            Object ctx, byte[] masterKey)
+                throws IOException, BookieException {
+        long requestNanos = MathUtils.nowInNano();
+        boolean success = false;
+        try {
+            LedgerDescriptor handle = handles.getHandle(ledgerId, masterKey);
+            synchronized (handle) {
+                if (handle.isFenced()) {
+                    throw BookieException
```

Comment:
Rethinking about fence/force I think you are right. While issuing a force() data is already on the bookie and so it is useless to break the writer. 
I thought it was a good idea because LAC on client advances only on force.
I am dropping this check